### PR TITLE
Fix put inference endpoint with adaptive allocations

### PR DESCRIPTION
**Pull request by jan-elastic**
The decision to disallow num_allocations + adaptive_allocations broke the put inference endpoint (num_allocations was mandatory, and then adaptive_allocations fails). That's fixed now.

### CODE REVIEW

**Review by davidkyle — APPROVED** [review]
LGTM

File: build-tools-internal/src/main/groovy/elasticsearch.run.gradle [hunk 1]
```diff
@@ -31,6 +31,11 @@ testClusters.register("runTask") {
       keystore 'bootstrap.password', 'password'
       user username: 'elastic-admin', password: 'elastic-password', role: '_es_test_root'
       numberOfNodes = 1
+
+      setting 'xpack.security.audit.enabled', 'true'
+      keystore 'telemetry.secret_token', 'fHWRkyJET2yd9DvS9m'
+      setting 'telemetry.metrics.enabled', 'true'
+      setting 'telemetry.agent.server_url', 'https://460d4c9cb692485ba3e4b6be78f8c378.apm.us-central1.gcp.cloud.es.io:443'
```
Comment (davidkyle) [inline build-tools-internal/src/main/groovy/elasticsearch.run.gradle:38]: Is this required? Looks like it's left from some other code you are working on

Comment (jan-elastic) [reply build-tools-internal/src/main/groovy/elasticsearch.run.gradle:38]: oops

Comment (jan-elastic) [reply build-tools-internal/src/main/groovy/elasticsearch.run.gradle]: removed

**Pull request merged**